element-ios/pipeline.yml

```diff
@@ -0,0 +1,58 @@
+# This is just a dummy entry (the `x-yaml-aliases` key is not an official pipeline key, and will be ignored by BuildKite)
+#   that we use only to store YAML anchors (`&xxx`), that we plan to use and reference later in the YAML file (using `*xxx`)
+#   without having to copy/paste the same values over and over.
+#   Note: keys like `agent`, `env`, … used here are totally arbitrary; the only point is to define various separate `&xxx` anchors there.
+#
+x-yaml-aliases:
+  agent: &macos
+    queue: "macos"
+    git: "github"
+  env: &env
+    FASTLANE_SKIP_UPDATE_CHECK: 1
+  conditions:
+    - &release_branch |
+      // Only on PR builds, and for branches starting with 'release/'
+      build.branch =~ /^release\// &&
+        build.env("NIGHTLY_BUILD") == null
+    - &feature_branch |
+      // Only on PR builds, and for branches other than 'develop', 'master', and 'release/*'
+      build.branch != 'master' &&
+        build.branch != 'develop' &&
+        build.branch !~ /^release\// &&
+        build.env("NIGHTLY_BUILD") == null
+    - &nightly |
+      // Only for nightly builds
+      build.env("NIGHTLY_BUILD") != null
+  commands:
+    - &setup |
+      brew install cmake
+      bundle install
+      printf %100s\\n | tr ' ' '='
+    - &build_app
+      bundle exec fastlane build
```

Comment:
I'm not convinced that using yaml anchors quite so widely is good for either readability or maintainability (imho you may as well write `bundle exec fastlane build` as `*build_app`, and only switch to yaml anchors when it gets more complicated), but I guess it's fine.

Reply:
yeah I might get overexcited with anchors ^^ it was quite useful when debugging the pipeline as it's then easy to change the command just here (to an `echo` for debug, or to run another fastlane lane, etc) in a single place and have it applied in both the steps for Feature branch and release branch.

That's also in preparation for future changes in these pipelines, as for now we do only a few steps (because the single MacMini CI machine that we have is not super powerful so we don't want to put it to its knees) but in the future we plan to add other steps, like building the same commands on master and develop branches too etc, so that will get repeated again when we introduce those.

Reply:
Turns out it was useful to factorize the `echo --- some section title` in each cases too after all 😜